src/appimagetool.c

```diff
@@ -142,7 +143,9 @@ int sfs_mksquashfs(char *source, char *destination, int offset) {
         gchar* offset_string;
         offset_string = g_strdup_printf("%i", offset);
 
-        char* args[32];
+        guint sqfs_opts_len = sqfs_opts ? g_strv_length(sqfs_opts) : 0;
+
+        char* args[32 + sqfs_opts_len];
```

Comment:
The `32` here always seemed arbitrary. Let's put a `TODO` above it, or fix this properly.

Reply:
I counted these manually and it looks like the max is len(sqfs_opts) + 22. I thought about adding a [G_STATIC_ASSERT()](https://www.freedesktop.org/software/gstreamer-sdk/data/docs/latest/glib/glib-Miscellaneous-Macros.html#G-STATIC-ASSERT:CAPS) But I guess this is not "static enough" for a compile time check.